src/app/footer/footer.component.spec.ts

```diff
@@ -1,28 +1,66 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
-import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { MatSnackBar, MatSnackBarModule } from '@angular/material/snack-bar';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { TranslateModule } from '@ngx-translate/core';
-import { Actions, NgxsModule } from '@ngxs/store';
+import { NgxsModule, Store } from '@ngxs/store';
 
+import { MessageBarComponent } from 'shared/components/message-bar/message-bar.component';
+import { SharedModule } from 'shared/shared.module';
+import { ClearMessageBar, ShowMessageBar } from 'shared/store/app.actions';
 import { FooterComponent } from './footer.component';
 
 describe('FooterComponent', () => {
   let component: FooterComponent;
   let fixture: ComponentFixture<FooterComponent>;
+  let store: Store;
+  let mockMatSnackBar: MatSnackBar;
+  const messagePayload = { message: 'test', type: 'success' };
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [MatSnackBarModule, NgxsModule.forRoot([]), TranslateModule.forRoot()],
+      imports: [
+        NoopAnimationsModule,
+        MatSnackBarModule,
+        SharedModule,
+        NgxsModule.forRoot([]),
+        TranslateModule.forRoot()
+      ],
       declarations: [FooterComponent]
     }).compileComponents();
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(FooterComponent);
     component = fixture.componentInstance;
+    store = TestBed.inject(Store);
+    mockMatSnackBar = TestBed.inject(MatSnackBar);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should show showSnackBar on action dispatch', () => {
+    jest.spyOn(component, 'showSnackBar');
+
+    component.showSnackBar(messagePayload);
+    store.dispatch([new ShowMessageBar(messagePayload), new ClearMessageBar()]);
```

Comment:
this line is actually useless, it won't call the `showSnackBar` method because we didn't call `ngOnInit`